examples/src/main/java/com/expediagroup/sdk/xap/examples/scenarios/lodging/ListingsHotelIdsSearchScenario.java

```diff
@@ -0,0 +1,430 @@
+
+  public static void main(String[] args) {
+    new ListingsHotelIdsSearchScenario().run();
+    System.exit(0);
+  }
+
+  @Override
+  public void run() {
+    LOGGER.info(
+        "======================== Running ListingsHotelIdsSearchScenario =======================");
+
+    List<String> propertyIds = getPropertyIdsFromDownloadUrl();
+    cachePropertyLocationFromDownloadUrl(propertyIds);
+    HotelListingsResponse hotelListingsResponse = getPropertyPriceFromLodgingListings(propertyIds);
+    displayResult(hotelListingsResponse);
+
+    LOGGER.info(
+        "========================== End ListingsHotelIdsSearchScenario =========================");
+  }
+
+  /**
+   * Retrieve accessible property ids from SDP DownloadURL API.
+   *
+   * @return property ids
+   */
+  private List<String> getPropertyIdsFromDownloadUrl() {
+    LOGGER.info(
+        "====================== Executing getPropertyIdsFromDownloadUrl =====================");
+
+    GetFeedDownloadUrlOperationParams getPropertyIdListParams =
+        GetFeedDownloadUrlOperationParams.builder()
+            // Use the type LISTINGS to get the list of accessible property ids.
+            .type(GetFeedDownloadUrlOperationParams.Type.LISTINGS)
+            // Without any filters, this operation will return the information of all lodging
+            // properties in en_US by default.
+            .build();
+
+    Response<PresignedUrlResponse> downloadUrlListingsResponse =
+        client.execute(new GetFeedDownloadUrlOperation(getPropertyIdListParams));
+
+    if (downloadUrlListingsResponse.getData() == null
+        || downloadUrlListingsResponse.getData().getBestMatchedFile() == null) {
+      throw new IllegalStateException("No listings file found");
+    }
+
+    // The download URL points to a zip file containing various jsonl files.
+    // Each line in the jsonl files contains a json object representing a property.
+    // For demonstration purposes, we will only read a few properties from the file without
+    // downloading the entire file.
+    String listingsDownloadUrl = downloadUrlListingsResponse.getData()
+        .getBestMatchedFile()
+        .getDownloadUrl();
+    LOGGER.info("Listings Download URL: {}", listingsDownloadUrl);
+
+    // Read property ids from the file.
+    List<String> propertyIds = getPropertyIdsFromListingsFile(listingsDownloadUrl);
+
+    if (propertyIds.isEmpty()) {
+      throw new IllegalStateException("No accessible property ids found.");
+    }
+    LOGGER.info("Accessible Property Ids: {}", propertyIds);
+
+    LOGGER.info(
+        "======================= getPropertyIdsFromDownloadUrl Executed ======================");
+    return propertyIds;
+  }
+
+  /**
+   * Cache the location content from SDP DownloadURL API.
+   *
+   * @param propertyIds The property ids that need the location content.
+   */
+  private void cachePropertyLocationFromDownloadUrl(List<String> propertyIds) {
```

Comment:
I am not sure if caching location info of property is a good idea, or shall we cache property name, description and thumbnail? Since they are more like the basic info of a property.

Reply:
I think the idea is to let the partners know that they can combine different SDP files as they need.

Specifically for this case, property name, description and heroImage can be found at the same time only in `SUMMARY` file. If we are using `LISTINGS` file to get property IDs in this case, then we would need to get the description name from `DESCRIPTIONS` file, thumbnail from `IMAGES` file and property name from another file, e.g. `LOCATIONS` file (because both `DESCRIPTIONS` and `IMAGES` file don't contain property name).